permits/models.py

```diff
@@ -1211,6 +1214,85 @@ class Meta:
         unique_together = [("property", "works_object_type_choice")]
 
 
+class PermitRequestComplementaryDocument(models.Model):
+    STATUS_TEMP = 0
+    STATUS_FINALE = 1
+    STATUS_OTHER = 2
+    STATUS_CANCELED = 3
+
+    STATUS_CHOICES = (
+        (STATUS_TEMP, _("Provisoire")),
+        (STATUS_FINALE, _("Final")),
+        (STATUS_OTHER, _("Autre")),
+        (STATUS_CANCELED, _("Annulé")),
+    )
+
+    document = models.FileField(_("Document"))
+    description = models.TextField(_("Description du document"), blank=True,)
+    owner = models.ForeignKey(
+        User,
+        null=True,
+        on_delete=models.SET_NULL,
+        verbose_name=_("Propriétaire du document"),
+    )
+    permit_request = models.ForeignKey(
+        PermitRequest,
+        null=False,
+        on_delete=models.CASCADE,
+        verbose_name=_("Demande de permis"),
+    )
+    status = models.PositiveSmallIntegerField(
+        _("Statut du document"), choices=STATUS_CHOICES,
+    )
+    authorised_departments = models.ManyToManyField(
+        PermitDepartment,
+        verbose_name=_("Département autorisé à visualiser le document"),
+        db_table=_("permits_authorised_departments"),
+    )
+    is_public = models.BooleanField(default=False, verbose_name=_("Public"))
+    document_type = models.ForeignKey(
+        "ComplementaryDocumentType",
+        null=True,
+        on_delete=models.SET_NULL,
+        verbose_name=_("Type du document"),
+    )
+
+    def delete(self, using=None, keep_parents=False):
+        # delete the uploaded file
+        os.remove(os.path.join(settings.MEDIA_ROOT, self.document.name))
```

Comment:
Catch `OSError` exceptions if there is a problem when removing the file, and raise `ProtectedError` with a message explaining why the Document can't be deleted.
https://docs.djangoproject.com/en/3.2/ref/exceptions/#django.db.models.ProtectedError